metagraph-plugin/metagraph_katana/plugins/metagraph_katana/algorithms.py

```diff
@@ -28,10 +33,12 @@ def kg_bfs_iter(graph: KatanaGraph, source_node: NodeID, depth_limit: int) -> Nu
     :rtype: NumpyVectorType
     '''
     bfs_prop_name = "bfs_prop_start_from_" + str(source_node)
+    depth_limit_internal = 2**30 - 1 if depth_limit == -1 else depth_limit # return all the reachable nodes for the default value of depth_limit (-1)
     start_node = source_node
-    bfs(graph.value, start_node, bfs_prop_name)
+    if not has_node_prop(graph.value, bfs_prop_name):
+        bfs(graph.value, start_node, bfs_prop_name)
     pg_bfs_list = graph.value.get_node_property(bfs_prop_name).to_pandas().values.tolist()
-    new_list = [[i, pg_bfs_list[i]] for i in range(len(pg_bfs_list)) if pg_bfs_list[i] < depth_limit]
+    new_list = [[i, pg_bfs_list[i]] for i in range(len(pg_bfs_list)) if pg_bfs_list[i] < depth_limit_internal]
```

Comment:
This didn't change. Can you please remove the arbitrary upper limit you are using for -1?